querybook/webapp/lib/query-result/detector.ts

```diff
@@ -54,6 +55,25 @@ const columnDetectors: IColumnDetector[] = [
     .concat(window.CUSTOM_COLUMN_DETECTORS ?? [])
     .sort((a, b) => b.priority - a.priority) as IColumnDetector[];
 
+export function getColumnTypesForTable(columns: string[], rows: any[][]) {
+    const sizeOfSample = Math.max(
+        DETECTOR_MIN_SAMPLE_SIZE,
+        Math.floor(rows.length / 100)
+    );
+    return columns.map((colName, index) => {
+        const notNullRowValues = rows.reduce((arr, row) => {
+            const value = row[index];
+            if (!isCellValNull(value)) {
+                arr.push(value);
+            }
+            return arr;
+        }, []);
```

Comment:
I think the previous way + filter will be more readable, `rows.map((row) => row[index]).filter((value) => !isCellValNull(value))`
and I think it will not impact the perf much.